groovy/postInit/chemistry/inorganic_chemistry/elements/p_block/group18/AirDistillation.groovy

```diff
@@ -14,411 +14,571 @@ FBR = recipemap('fixed_bed_reactor')
-    .fluidOutputs(fluid('subcooled_oxygen_rich_liquid') * 75)
-    .fluidOutputs(fluid('nitrogen') * 9600)
+    
+//KRYPTON & XENON
+SINGLE_COLUMN_CRYOGENIC_DISTILLATION_PLANT.recipeBuilder()
+    .fluidInputs(fluid('untreated_liquid_oxygen') * 105)
+    .fluidOutputs(fluid('krypton_xenon_rich_liquid')) // Contains 0.0264 Kr and 0.00216 Xe gas
+    .fluidOutputs(fluid('treated_liquid_oxygen') * 105)
     .duration(5)
+    .EUt(Globals.voltAmps[1])
     .buildAndRegister()
 
-//NORMAL SUBCOOLING
-HEAT_EXCHANGER.recipeBuilder()
-    .fluidInputs(fluid('oxygen_rich_liquid') * 75)
-    .fluidInputs(fluid('liquid_argon_product') * 5)
-    .fluidOutputs(fluid('subcooled_oxygen_rich_liquid') * 75)
-    .fluidOutputs(fluid('partially_liquefied_argon') * 18)
+BR.recipeBuilder()
+    .fluidInputs(fluid('krypton_xenon_rich_liquid'))
+    .fluidInputs(fluid('oxygen') * 10)
+    .fluidOutputs(fluid('combusted_krypton_xenon_concentrate') * 48)
     .duration(5)
+    .EUt(Globals.voltAmps[1])
     .buildAndRegister()
 
-PHASE_SEPARATOR.recipeBuilder()
-    .fluidInputs(fluid('partially_liquefied_argon') * 450)
-    .fluidOutputs(fluid('argon') * 400)
-    .fluidOutputs(fluid('liquid_argon_product') * 50)
-    .duration(29)
+PSA.recipeBuilder()
+    .notConsumable(metaitem('zeolite_membrane'))
+    .fluidInputs(fluid('combusted_krypton_xenon_concentrate') * 48)
+    .fluidOutputs(fluid('purified_krypton_xenon_concentrate') * 48)
+    .duration(100)
     .EUt(Globals.voltAmps[3])
     .buildAndRegister()
 
-//ADAPTED LOW PRESSURE RECIPE
-LOW_PRESSURE_DISTILLATION_TOWER.recipeBuilder()
-    .circuitMeta(3)
-    .fluidInputs(fluid('oxygen_rich_liquid') * 100)
-    .fluidInputs(fluid('oxygen_rich_gas') * 1600)
-    .fluidInputs(fluid('nitrogen_rich_gas') * 6400)
-    .fluidInputs(fluid('subcooled_liquid_nitrogen') * 100)
-    .fluidOutputs(fluid('untreated_liquid_oxygen') * 105)
-    .fluidOutputs(fluid('argon_rich_gas') * 400)
-    .fluidOutputs(fluid('cold_waste_gaseous_nitrogen') * 1280)
-    .fluidOutputs(fluid('liquid_nitrogen_product') * 300)
-    .duration(5)
+SINGLE_COLUMN_CRYOGENIC_DISTILLATION_PLANT.recipeBuilder()
+    .fluidInputs(fluid('purified_krypton_xenon_concentrate') * 22224)
+    .fluidOutputs(fluid('krypton') * 12)
+    .chancedFluidOutput(fluid('krypton'), 2222, 0)
+    .fluidOutputs(fluid('liquid_crude_xenon'))
+    .duration(100)
     .EUt(Globals.voltAmps[3])
     .buildAndRegister()
 
-//ADAPTED HEAT EXCHANGE (SUBCOOLING)
-HEAT_EXCHANGER.recipeBuilder()
-    .circuitMeta(1)
-    .fluidInputs(fluid('cold_waste_gaseous_nitrogen') * 1280)
-    .fluidInputs(fluid('untreated_liquid_nitrogen') * 100)
-    .fluidOutputs(fluid('chilly_waste_gaseous_nitrogen') * 1280)
-    .fluidOutputs(fluid('subcooled_liquid_nitrogen') * 100)
-    .duration(5)
+BR.recipeBuilder()
+    .fluidInputs(fluid('liquid_crude_xenon'))
+    .fluidInputs(fluid('oxygen') * 10)
+    .fluidOutputs(fluid('oxidized_xenon'))
+    .duration(100)
```

Comment:
I mean, I know this is in series, but it does look a little off to have this recipe and the next one take 5 seconds each to process a single liter of xenon.

Reply:
Actually, is this an issue based on not accounting for the change in volume between liquid crude xenon and oxidized xenon (assuming it becomes a gas)?

Reply:
no there's just so little xenon in the air that you get like 0.0216 xenon per 24000L of air so any larger recipes would take ages to fill up